src/main/scala/io/iohk/ethereum/jsonrpc/FilterManager.scala

```diff
@@ -0,0 +1,253 @@
+package io.iohk.ethereum.jsonrpc
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.util.{ByteString, Timeout}
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.domain._
+import io.iohk.ethereum.jsonrpc.EthService.BlockParam
+import io.iohk.ethereum.keystore.KeyStore
+import io.iohk.ethereum.ledger.BloomFilter
+import io.iohk.ethereum.transactions.PendingTransactionsManager
+
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Random
+
+class FilterManager(
+    blockchain: Blockchain,
+    appStateStorage: AppStateStorage,
+    keyStore: KeyStore,
+    pendingTransactionsManager: ActorRef)
+  extends Actor {
+
+  import FilterManager._
+  import akka.pattern.{ask, pipe}
+
+  val maxBlockHashesChanges = 256
+
+  var filters: Seq[Filter] = Nil
```

Comment:
:+1: 

And do you think it's worthy to join `filters` and `lastCheckBlocks` within the same `Map`?